arcgis-ios-sdk-samples/Utility network/Create load report/CreateLoadReportViewController.swift

```diff
@@ -0,0 +1,360 @@
+            startingLocation.terminal = assetType.terminalConfiguration?.terminals.first(where: { $0.name == terminalName })
+            return startingLocation
+        } else {
+            return nil
+        }
+    }
+    
+    /// Get the utility tier's trace configuration.
+    func getTraceConfiguration() -> AGSUtilityTraceConfiguration? {
+        // Get a default trace configuration from a tier in the network.
+        utilityNetwork
+            .definition
+            .domainNetwork(withDomainNetworkName: "ElectricDistribution")?
+            .tier(withName: "Medium Voltage Radial")?
+            .traceConfiguration
+    }
+    
+    // MARK: Actions
+    
+    @IBAction func resetBarButtonItemTapped(_ sender: UIBarButtonItem) {
+        summaries.removeAll()
+        tableView.reloadSections([Section.included.rawValue], with: .automatic)
+        resetBarButtonItem.isEnabled = false
+        runBarButtonItem.isEnabled = !includedPhases.isEmpty
+    }
+    
+    @IBAction func runBarButtonItemTapped(_ sender: UIBarButtonItem) {
+        runBarButtonItem.isEnabled = false
+        SVProgressHUD.show(withStatus: "Creating load report…")
+        
+        let traceGroup = DispatchGroup()
+        for phase in includedPhases {
+            guard let phaseCode = phase.code else { continue }
+            // Create a conditional expression.
+            let phasesAttributeComparison = AGSUtilityNetworkAttributeComparison(networkAttribute: phasesNetworkAttribute, comparisonOperator: .doesNotIncludeAny, value: phaseCode)!
+            // Chain it with the base condition using an OR operator.
+            traceParameters.traceConfiguration?.traversability?.barriers = AGSUtilityTraceOrCondition(leftExpression: initialExpression, rightExpression: phasesAttributeComparison)
+            
+            traceGroup.enter()
+            utilityNetwork.trace(with: traceParameters) { [weak self] results, _ in
+                defer { traceGroup.leave() }
+                // Return if not result and ignore any trace error.
+                guard let self = self, let results = results else { return }
+                var totalCustomers = 0
+                var totalLoad = 0
+                results.forEach { result in
+                    switch result {
+                    case let elementResult as AGSUtilityElementTraceResult:
+                        // Get the unique customers count.
+                        totalCustomers = Set(elementResult.elements.map(\.objectID)).count
+                    case let functionResult as AGSUtilityFunctionTraceResult:
+                        // Get the total load with a function output.
+                        totalLoad = functionResult.functionOutputs.first?.result as? Int ?? 0
+                    default:
+                        break
+                    }
+                }
+                self.summaries[phase] = PhaseSummary(totalCustomers: totalCustomers, totalLoad: totalLoad)
```

Comment:
As I've already mentioned, I recommend have a local `summaries` variable for accumulating the new summaries and then using it to update the `summaries` property when you have all of them.